fiftyone/core/dataset.py

```diff
@@ -223,8 +223,13 @@ def name(self):
 
     @name.setter
     def name(self, name):
-        self._doc.name = name
-        self._doc.save()
+        _name = self._doc.name
```

Comment:
@tylerganter there must be a more stylistic pattern for reverting doc changes if a `save()` fails. Any ideas?

Reply:
This gets into database transactions, which I don't think is important right now if this works

Reply:
👍

Reply:
@brimoor the one alternative could be:
```python
try:
    self._doc.name = name
    self._doc.save()
except:
    self._doc.reload()
    raise
```
but I don't think you want to do that because it could automatically wipe any other changes. This is a bit of a special case where we for sure want to revert the change locally if it fails when writing, so I think what you have here is fine